MantidQt/CustomInterfaces/src/Reflectometry/ReflSaveTabPresenter.cpp

```diff
@@ -0,0 +1,181 @@
+    filterWorkspaceNames(m_view->getFilter(), m_view->getRegexCheck());
+    break;
+  case workspaceParamsFlag:
+    populateParametersList(m_view->getCurrentWorkspaceName());
+    break;
+  case saveWorkspacesFlag:
+    saveWorkspaces(m_view->getSavePath());
+    break;
+  }
+}
+
+/** Fills the 'List of Workspaces' widget with the names of all available
+* workspaces
+*/
+void ReflSaveTabPresenter::populateWorkspaceList() {
+  m_view->clearWorkspaceList();
+  m_view->setWorkspaceList(getAvailableWorkspaceNames());
+}
+
+/** Filters the names in the 'List of Workspaces' widget
+* @param filter :: filter text string
+* @param regexCheck :: whether to use regex in filtering
+*/
+void ReflSaveTabPresenter::filterWorkspaceNames(std::string filter,
+                                                bool regexCheck) {
+  m_view->clearWorkspaceList();
+  auto wsNames = getAvailableWorkspaceNames();
+  std::vector<std::string> validNames(wsNames.size());
+  auto it = validNames.begin();
+
+  if (regexCheck) {
+    // Use regex search to find names that contain the filter sequence
+    try {
+      boost::regex rgx(filter);
+      it = std::copy_if(
+          wsNames.begin(), wsNames.end(), validNames.begin(),
+          [rgx](std::string s) { return boost::regex_search(s, rgx); });
+    } catch (boost::regex_error &) {
+      g_log.error("Error, invalid regular expression\n");
+    }
+  } else {
+    // Otherwise simply add names where the filter string is found in
+    it = std::copy_if(wsNames.begin(), wsNames.end(), validNames.begin(),
+                      [filter](std::string s) {
+                        return s.find(filter) != std::string::npos;
+                      });
+  }
+
+  validNames.resize(std::distance(validNames.begin(), it));
+  m_view->setWorkspaceList(validNames);
+}
+
+/** Fills the 'List of Logged Parameters' widget with the parameters of the
+* currently selected workspace
+* @param wsName :: name of selected workspace
+*/
+void ReflSaveTabPresenter::populateParametersList(std::string wsName) {
+  m_view->clearParametersList();
+
+  std::vector<std::string> logs;
+  const auto &properties = AnalysisDataService::Instance()
+                               .retrieveWS<MatrixWorkspace>(wsName)
+                               ->run()
+                               .getProperties();
+  for (auto it = properties.begin(); it != properties.end(); it++) {
+    logs.push_back((*it)->name());
+  }
+  m_view->setParametersList(logs);
+}
+
+/** Saves selected workspaces
+*/
+void ReflSaveTabPresenter::saveWorkspaces(std::string saveDir) {
+  // Check that save directory is valid
+  if (saveDir.empty() || Poco::File(saveDir).isDirectory() == false) {
+    g_log.error("Directory specified doesn't exist or was invalid for your "
+                "operating system");
+    return;
+  }
+
+  // Create the appropriate save algorithm
+  bool titleCheck = m_view->getTitleCheck();
+  auto wsNames = m_view->getSelectedWorkspaces();
```

Comment:
I got confused when performing functional testing, because I was clicking on `Save` but forgot to select the workspaces to save. Could we add a warning/error message when `wsNames` is empty? Something like "No workspaces selected. You must select the workspaces to save" or something like that. I think the old interfaces saves the first workspace in the list in this case.